docs/common/craft-parts/reference/plugins/go_plugin.rst

```diff
@@ -0,0 +1,95 @@
+.. _craft_parts_go_plugin:
+
+Go plugin
+=========
+
+The Go plugin builds `Go`_ modules, which are collections of packages stored
+in a file tree containing a ``go.mod`` file at the root. After a successful
+build, this plugin will install the generated binaries in
+``$CRAFT_PART_INSTALL/bin``.
+
+
+Keywords
+--------
+
+In addition to the common :ref:`plugin <part-properties-plugin>` and
+:ref:`sources <part-properties-sources>` keywords, this plugin provides the following
+plugin-specific keywords:
+
+go-buildtags
+~~~~~~~~~~~~
+**Type:** list of strings
+**Default:** []
+
+`Build tags`_ to use during the build. The default behavior is not to use any
+build tags.
+
+go-generate
+~~~~~~~~~~~
+**Type:** list of strings
+**Default:** []
+
+Parameters to pass to `go generate`_ before building. Each item on the list
+will be a separate ``go generate`` call. The default behavior is not to call
+``go generate``.
+
+Environment variables
+---------------------
+
+During build, this plugin sets ``GOBIN`` to ``${CRAFT_PART_INSTALL}/bin``.
+
+.. _go-details-begin:
+
+Dependencies
+------------
+
+The Go plugin needs the ``go`` executable to build Go programs but does not
+provision it by itself, to allow flexibility in the choice of compiler version.
+
+Common means of providing ``go`` are:
+
+* The ``golang`` Ubuntu package, declared as a ``build-package``.
+* The ``go`` snap, declared as a ``build-snap`` from the desired channel.
+
+Another alternative is to declare another part with the name ``go-deps``. In
```

Comment:
May be useful to note that `after: [go-deps]` should be added to any parts using the go plugin.